### PR TITLE
Refactor changed locale files detection

### PR DESCRIPTION
**Pull request by mathjazz**
While working on #440, I also noticed a few nits to make code slighlty more readable.

@jotes r?

### CODE REVIEW

File: pontoon/sync/tasks.py [hunk 1]
```diff
@@ -173,10 +173,12 @@ def sync_project_repo(self, project_pk, repo_pk, project_sync_log_pk, now, obsol
         repo_sync_log.end()
         return
 
+    obsolete_entities_paths = Resource.objects.obsolete_entities_paths(obsolete_vcs_entities) if obsolete_vcs_entities else None
+
     vcs_project = VCSProject(
         db_project,
         locales=locales,
-        obsolete_entities_paths=Resource.objects.obsolete_entities_paths(obsolete_vcs_entities),
+        obsolete_entities_paths=obsolete_entities_paths,
```
Comment (mathjazz) [inline pontoon/sync/tasks.py]: Without this, `sync_project`fails is ran with the `locale` flag.

File: pontoon/sync/tasks.py [hunk 2]
```diff
@@ -250,6 +266,13 @@ def sync_project_repo(self, project_pk, repo_pk, project_sync_log_pk, now, obsol
                 if not no_commit and locale in changeset.locales_to_commit:
                     commit_changes(db_project, vcs_project, changeset, locale)
 
+                log.info(
+                    'Synced locale {locale} for project {project}.'.format(
+                        locale=locale.code,
+                        project=db_project.slug,
+                    )
+                )
+
```
Comment (mathjazz) [inline pontoon/sync/tasks.py]: Adding a few more log messages shouldn't hurt. If syncing a project with many locales, we only printed log messages if there was something to commit before the final `Synced translations for project {0} in locales {1}` log message.

Comment (jotes) [inline pontoon/sync/tasks.py]: I agree 😊

**Comment by jotes** [comment]
r+

**Pull request merged**